namespaces/mson-namespace.md

```diff
@@ -0,0 +1,503 @@
+
+## Number Type (JSON:Number Type)
+
+- Include MSON Element
+
+## Array Type (JSON:Array Type)
+
+- Include MSON Element
+
+## Object Type (JSON:Object Type)
+
+- Include MSON Element
+
+## Property Type (JSON:Property Type)
+
+- Include MSON Element
+
+## Enum Type (MSON Element)
+
+Enumeration type. Exclusive list of possible elements. The elements in content's array MUST be interpreted as mutually exclusive.
+
+### Properties
+
+- `element`: enum (string, required, fixed)
+- `content` (array[MSON Element])
+
+### Example
+
+#### MSON
+
+```
+- tag (enum[string])
+    - red
+    - green
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "property",
+            "attributes": {
+                "name": "tag"
+            },
+            "content": {
+                "element": "enum",
+                "content": [
+                    {
+                        "element": "string",
+                        "content": "red"
+                    },
+                    {
+                        "element": "string",
+                        "content": "green"
+                    }
+                ]
+            }
+        }
+    ]
+}
+```
+
+
+
+## Examples
+
+### Anonymous Object Type
+
+#### MSON
+
+```
+- id: 1
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "property",
+            "attributes": {
+                "name": "id"
+            },
+            "content": {
+                "element": "string",
+                "content": "1"
+            }
+        }
+    ]
+}
+```
+
+### Type Attributes
+
+#### MSON
+
+```
+- id: 42 (required, fixed)
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "property",
+            "attributes": {
+                "name": "id",
+                "typeAttributes": ["required", "fixed"]
+            },
+            "content": {
+                "element": "string",
+                "content": "42"
+            }
+        }
+    ]
+}
+```
+
+### Default Value
+
+#### MSON
+
+```
+- id (number)
+    - default: 0
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "property",
+            "attributes": {
+                "name": "id",
+                "default": {
+                    "element": "number",
+                    "content": 0
+                }
+            },
+            "content": {
+                "element": "number",
+                "content": null
+            }
+        }
+    ]
+}
+```
+
+### One Of
+
+#### MSON
+
+```
+- city
+- One Of
+    - state
+    - province
+```
+
+#### MSON DOM
+
+```json
+{
+    "element": "object",
+    "content": [
+        {
+            "element": "property",
+            "attributes": {
+                "name": "city"
+            }
+        },
+        {
+            "element": "select",
```

Comment:
I understand this, but in the spec, there is no where any mention of `content` containing `Select` named type.